src/Components/CommentCreator.jsx

```diff
@@ -0,0 +1,90 @@
+import { useState } from "react"
+import { postComment } from "../utils/api"
+import { DateFormatter } from "./DateFormatter"
+
+export const CommentCreator = ({ review_id }) => {
+
+
+    // NEED TO CHANGE COMMENT ORDER SO MOST RECENT COMES FIRST!!
+
+    const [username, setUsername] = useState("")
+    const [commentPosted, setCommentPosted] = useState(false)
+    const [isLoading, setIsLoading] = useState(false)
+    const [commentText, setCommentText] = useState("")
+    const [disabled, setDisabled] = useState(false)
+    const [newComment, setNewComment] = useState({})
+    const [isError, setIsError] = useState(false)
+
+    const handleSubmit = (event) => {
+        event.preventDefault();
+        setIsError(false)
+        setIsLoading(true)
+        setDisabled(true)
+        setCommentPosted(false)
+
+        postComment(review_id, username, commentText)
+        .then((comment) => {
+            setCommentPosted(true);
+            setIsLoading(false)
+            setUsername("")
+            setCommentText("")
+            setNewComment(comment)
+        })
+        .catch((error) => {
+            setIsLoading(false)
+            setIsError(true)
+            setDisabled(false)
+            })
+    }
+
+    if(isError){
+        return(<p>"There was an error posting your comment, please ensure your username is correct and try again!"</p>)
+    }
+
+    if(isLoading){
+        return(<p>"Loading..."</p>)
+    }
+
+    if(commentPosted){
+        return(
+        
+        <section>
+            <p>Your comment has been posted: </p>
```

Comment:
More of a design choice but I would think about changing the colour as this looks like part of the comment.